lib/http/2/stream.rb

```diff
@@ -587,6 +592,13 @@ def process_priority(frame)
       #   We should add code here.
     end
 
+    def adjust_recv_window_size(delta_size)
+      if @recv_window_size > @local_window || @recv_window_size > 0x7fffffff - delta_size
+        stream_error(:flow_control_error)
```

Comment:
Where did case come from? It's not tested and it's not clear how or why this would be triggered?

Reply:
This was directly inherited from [here]( https://github.com/nghttp2/nghttp2/blob/2bf3680d870953010d7e1e6e4a66510f8458cc3c/lib/nghttp2_session.c#L4904-L4922) and [here](https://github.com/nghttp2/nghttp2/blob/2bf3680d870953010d7e1e6e4a66510f8458cc3c/lib/nghttp2_session.c#L4944). 

But maybe, if not counting with recv_window_size, this check is being handled somewhere else?